e2e/machine_migration_test.go

```diff
@@ -111,6 +129,221 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+				Eventually(func() error {
+					// Get the latest version again right before update
+					currentMachine, err := mapiframework.GetMachine(cl, testMachineName)
+					if err != nil {
+						return err
+					}
+
+					updatedMachine := currentMachine.DeepCopy()
+					updatedMachine.Spec.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
+
+					err = cl.Update(ctx, updatedMachine)
+					return err
+				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Should allow modification of authoritativeAPI")
+
+				// Verify the machine still exists after the change (don't wait for specific status)
+				Eventually(func() error {
+					_, err := mapiframework.GetMachine(cl, testMachineName)
+					return err
+				}, capiframework.WaitLong, capiframework.RetryLong).Should(Succeed(), "Machine should still exist after authoritativeAPI change")
+
+				// Clean up the test machine
+				DeferCleanup(func() {
+					By("Cleaning up test machine")
+					// Try to delete the MAPI machine first
+					if testMachine != nil {
+						mapiframework.DeleteMachines(ctx, cl, testMachine)
+					}
+					// Try to delete the CAPI machine as well
+					capiMachine := &clusterv1.Machine{}
+					err := cl.Get(ctx, client.ObjectKey{Name: testMachineName, Namespace: capiframework.CAPINamespace}, capiMachine)
+					if err == nil {
+						cl.Delete(ctx, capiMachine)
+					}
+					// Don't wait for deletion to complete - just attempt it
+				})
+			})
+		})
+
+		Context("with spec.authoritativeAPI: ClusterAPI, Prevent changes to non-authoritative Machines except from sync controller;", func() {
+			var testMachineName = "machine-vap-83955"
+			var testMapiMachine *machinev1beta1.Machine
+			var testCapiMachine *clusterv1.Machine
+
+			BeforeAll(func() {
+				testMapiMachine = createMAPIMachineWithAuthority(ctx, cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+
+				DeferCleanup(func() {
+					By("Cleaning up machine resources")
+					cleanupMachineResources(
+						ctx,
+						cl,
+						[]*clusterv1.Machine{testCapiMachine},
+						[]*machinev1beta1.Machine{testMapiMachine},
+					)
+				})
+			})
+
+			It("should verify CAPI Machine gets created and becomes Running", func() {
+				verifyMachineRunning(cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+			})
+
+			It("should verify that the non-authoritative MAPI Machine has an authoritative CAPI Machine mirror", func() {
+				verifyCAPIMachineExists(cl, testMachineName)
+			})
+
+			It("should prevent modification of InstanceType in non-authoritative MAPI Machine", func() {
+				verifyFieldModificationPrevented(cl, testMapiMachine.Name, "InstanceType", func(machine *machinev1beta1.Machine) {
+					err := updateMachineProviderSpec(machine, func(providerSpec *machinev1beta1.AWSMachineProviderConfig) {
+						providerSpec.InstanceType = testInstanceType
+					})
+					if err != nil {
+						Fail(fmt.Sprintf("Failed to update provider spec: %v", err))
```

Comment:
This would be more consistent with existing code / repo standards :) 

```suggestion
						Fail(fmt.Sprintf("Failed to update provider spec: %v", err))
						 Expect(err).NotTo(HaveOccurred(), "Failed to get current machine")

```

Reply:
Also no need for the `if err != nil {` then :-)